tests/periph_uart/main.c

```diff
@@ -57,16 +57,97 @@ static int parse_dev(char *arg)
 {
     unsigned dev = atoi(arg);
     if (dev >= UART_NUMOF) {
-        printf("Error: Invalid UART_DEV device specified (%u).\n", dev);
+        printf("msg: Invalid UART_DEV device specified (%u).\n", dev);
         return -1;
     }
     else if (UART_DEV(dev) == STDIO_UART_DEV) {
-        printf("Error: The selected UART_DEV(%u) is used for the shell!\n", dev);
+        printf("msg: The selected UART_DEV(%u) used for the shell!\n", dev);
         return -2;
     }
     return dev;
 }
 
+#ifdef MODULE_PERIPH_UART_MODECFG
+static uart_data_bits_t parse_data_bits(char *arg)
+{
+    int data_bits = atoi(arg);
+    uart_data_bits_t res;
+
+    switch (data_bits) {
+        case 5:
+            res = UART_DATA_BITS_5;
+            break;
+        case 6:
+            res = UART_DATA_BITS_6;
+            break;
+        case 7:
+            res = UART_DATA_BITS_7;
+            break;
+        case 8:
+            res = UART_DATA_BITS_8;
+            break;
+        default:
+            printf("msg: Invalid number of data_bits (%i).\n", data_bits);
+            res = -1;
+    }
+
+    return res;
+}
+
+static uart_stop_bits_t parse_stop_bits(char *arg)
+{
+    int stop_bits = atoi(arg);
+    uart_stop_bits_t res;
+
+    switch (stop_bits) {
+        case 1:
+            res = UART_STOP_BITS_1;
+            break;
+        case 2:
+            res = UART_STOP_BITS_2;
+            break;
+        default:
+            printf("msg: Invalid number of stop bits (%i).\n", stop_bits);
+            res = -1;
+    }
+
+    return res;
+}
+
+static uart_parity_t parse_parity(char *arg)
+{
+    uart_parity_t res;
+
+    switch (arg[0]) {
+        case 'n':
```

Comment:
ne need for the 'double cases', simply substract the ASCII offeset before hand:
```c
if (arg[0] > 'Z') {
  arg[0] -= 32;
}

switch (arg[0]) {
   case 'N': ...
```

Reply:
even better :-)

Reply:
Done